posthog/models/async_deletion/delete_events.py

```diff
@@ -45,40 +46,37 @@ def process(self, deletions: list[AsyncDeletion]):
             },
         )
 
-        conditions, args = self._conditions(deletions)
+        conditions, args = [], {}
+        for i, deletion in enumerate(deletions):
+            condition, arg = self._condition(deletion, str(i))
 
-        # Split the deletions into chunks to avoid hitting the max query size
-        query_predicate = []
-        for condition in conditions:
-            query_predicate.append(condition)
+            conditions.append(condition)
+            args.update(arg)
 
             # Get estimated  byte size of the query
-            str_predicate = " OR ".join(query_predicate)
-            query_size = len(str_predicate.encode("utf-8"))
+            str_predicate = " OR ".join(conditions)
+            query = f"DELETE FROM sharded_events ON CLUSTER '{CLICKHOUSE_CLUSTER}' WHERE {str_predicate}"
+            query_size = len(query.encode("utf-8"))
+
+            logger.debug(f"Query size: {query_size}")
+            logger.debug(f"Query: {query}")
+            logger.debug(f"Query deletions: {deletions}")
 
             # If the query size is greater than the max predicate size, execute the query and reset the query predicate
             if query_size > MAX_PREDICATE_SIZE:
```

Comment:
Not sure what the maximum query length is offhand or how close this bumps into it so I'm not sure if this is useful or not, but it seems like it'd be safer if the query exceeds the predicate size to use the query from the _last_ iteration, not this one _after_ the limit is exceeded?